modules/ding_place2book/lib/p2b/P2b.php

```diff
@@ -0,0 +1,636 @@
+      ':event_id' => $event_id,
+      ':price_id' => $price_id,
+    ];
+    $url = $this->p2bFormatUrl(self::$updatePrice, $placeholders);
+    $result = $this->p2bRequest($url, 'PUT', $data);
+
+    return $result;
+  }
+
+  /**
+   * Request to p2b for delete price.
+   *
+   * @param string $event_maker_id
+   *   String that represents eventMaker id on p2b service.
+   * @param string $event_id
+   *   String that represents event id on p2b service.
+   * @param string $price_id
+   *   String that represents price id id on p2b service.
+   *
+   * @return array
+   *   Return object with data related to event.
+   *
+   * @throws \Exception
+   *   In case when not all required params was given.
+   *
+   * @see http://developer.place2book.com/v1/prices/
+   */
+  public function deletePrice($event_maker_id, $event_id, $price_id) {
+    if (empty($event_id) || empty($event_maker_id) || empty($price_id)) {
+      throw new \Exception("Params event_id, event_maker_id and required are required. Was given event_maker_id: {$event_maker_id}, event_id: {$event_id}, price_id: {$price_id}.");
+    }
+
+    $placeholders = [
+      ':event_maker_id' => $event_maker_id,
+      ':event_id' => $event_id,
+      ':price_id' => $price_id,
+    ];
+    $url = $this->p2bFormatUrl(self::$deletePrice, $placeholders);
+    $result = $this->p2bRequest($url, 'DELETE', [], 204);
+
+    return $result->price;
+  }
+
+  /**
+   * Main method. Make a http request.
+   *
+   * @param string $url
+   *   String that represents url to p2b service.
+   * @param string $type
+   *   String that represents type of HTTP request - GET/POST.
+   * @param array $params
+   *   Array with addition params for HTTP request.
+   * @param int $code
+   *   Correct code of response.
+   *
+   * @return object If as result got FALSE.
+   *   If as result got FALSE.
+   *
+   * @throws \P2b\Exception
+   *   In case when response code not equal to excepted.
+   */
+  private function p2bRequest($url, $type, array $params = [], $code = 200) {
+    $options = [];
+    $options['headers'] = [
+      "X-PLACE2BOOK-API-TOKEN" => self::$token,
+      "Accept" => "application/vnd.place2book.v1+json",
+    ];
+
+    if (!empty($params)) {
+      if ($type == 'POST') {
+        $options['json'] = $params;
+      }
+      else {
+        $options['query'] = $params;
+      }
+    }
+
+    $options['http_errors'] = FALSE;
+
+    $response = self::$client->request($type, $url, $options);
+    if ($response->getStatusCode() == $code) {
+      $data = $this->parseResponse($response->getBody());
+    }
+    else {
+      throw new Exception('Error. Wrong code was returned.');
+    }
+
+    return $data;
+  }
+
+  /**
+   * Parse json response from service.
+   *
+   * @param string $response
+   *   Json response from service.
+   *
+   * @return object
+   *   Parsed reponse as object.
+   */
+  private function parseResponse($response) {
+    $data = json_decode($response);
```

Comment:
Note that `json_decode` might return `NULL`. The result of this decoding method in some places is directly passed to iterators, like `foreach`, therefore an error will be triggered.